src/Framework/FieldsAPI/WpEditor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Give\Framework\FieldsAPI;
+
+/**
+ * @unreleased
+ */
+class WpEditor extends Field {
+	use Concerns\HasEmailTag;
+	use Concerns\HasHelpText;
+	use Concerns\HasLabel;
+	use Concerns\ShowInReceipt;
+	use Concerns\StoreAsMeta;
+
+	/**
+	 * @var string
+	 */
+	const TYPE = 'wp-editor';
+
+	/**
+	 * support: teeny, rich ( without media and quick tags )
+	 *
+	 * @var string
+	 */
+	protected $editorType = 'teeny';
+
+	/**
+	 * @see wp_editor settings: https://developer.wordpress.org/reference/classes/_wp_editors/parse_settings/
+	 *
+	 * @var array
+	 */
+	protected $editorConfig;
+
+	/**
+	 * @unreleased
+	 *
+	 * @param bool $editorType
+	 *
+	 * @return $this
+	 */
+	public function richTextEditorType( $editorType ) {
+		$this->editorType = $editorType;
+
+		return $this;
+	}
```

Comment:
No, we want to avoid these kinds of methods wherein the user just sets an internal string. This puts too much hope on the user to use the correct, enumerated strings. I'm also not sure we need this since the person can set the whole config, anyway. Why call out this specific property?

My suggestion is to set teeny to default in the `getEditorConfig` method and let the user override it that way, if they want.

Lastly, the irony isn't lost on me that I'm saying not to let them set a string, but they can set a whole config array. But that's really the nature of this field — which I'm really hoping won't get much use.

Reply:
@JasonTheAdams  This is good thought. I added abstraction to set editor type: https://github.com/impress-org/givewp/compare/ddc803ea03ad...d83d88a28a86

Let me know what do you think